designer/client/components/toolbars/test/buttons/TestWithSchemaButton.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  getProcessId, getProcessToDisplay,
+  getTestCapabilities, getTestViewParameters,
+  isLatestProcessVersion
+} from "../../../../reducers/selectors/graph"
+import {useWindows, WindowKind} from "../../../../windowManager"
+import {ToolbarButtonProps} from "../../types"
+import ToolbarButton from "../../../toolbarComponents/ToolbarButton";
+import _, {set} from "lodash"
+import {TestViewParameters} from "../../../../common/TestResultUtils";
+import {testProcessFromJson} from "../../../../actions/nk/displayTestResults";
+import {expressionStringToJsonType} from "../TestWithSchemaUtils";
+import {GenericActionParameters} from "../../../modals/GenericActionDialog";
+
+type Props = ToolbarButtonProps
+
+function TestWithSchemaButton(props: Props) {
+  const {disabled} = props
+  const {t} = useTranslation()
+  const {open} = useWindows()
+  const processIsLatestVersion = useSelector(isLatestProcessVersion)
+  const testCapabilities = useSelector(getTestCapabilities)
+  const testViewParameters: TestViewParameters[] = useSelector(getTestViewParameters)
+  const processId = useSelector(getProcessId)
+  const processToDisplay = useSelector(getProcessToDisplay)
+  const dispatch = useDispatch()
+
+  const available = !disabled && processIsLatestVersion && testCapabilities && testCapabilities.canCreateTestView
+
+  const [action, setAction] = useState(null)
+  const [selectedSource, setSelectedSource] = useState(_.head(testViewParameters)?.sourceId)
+  const [sourceParameters, setSourceParameters] = useState(updateParametersFromTestView())
+
+  function updateParametersFromTestView(): {[key: string]: GenericActionParameters} {
+    return (testViewParameters || []).reduce((testViewObj, testViewParam) => ({
+      ...testViewObj,
+      [testViewParam.sourceId]: {
+        parameters: testViewParam.parameters,
+        parametersValues: (testViewParam.parameters || []).reduce((paramObj, param) => ({
+          ...paramObj,
+          [param.name]: param.defaultValue,
+        }), {}),
+        onParamUpdate: (name: string) => (value: any) => onParamUpdate(testViewParam.sourceId, name, value)
+      }
+    }), {})
+  }
+
+  function onParamUpdate(sourceId: string, name: string, value: any) {
+    setSourceParameters(current => ({
+      ...current,
+      [sourceId]: {
+        ...current[sourceId],
+        parametersValues: {
+          ...current[sourceId].parametersValues,
+          [name]: {expression: value, language: current[sourceId].parametersValues[name].language}
+        }
+      }
+    }))
+  }
+
+  const onConfirmAction = useCallback((paramValues) => {
+    const recordJson = {}
+    sourceParameters[selectedSource].parameters.forEach(uiParam => {
+      set(recordJson, uiParam.name, expressionStringToJsonType(paramValues[uiParam.name].expression, uiParam.typ))
+    })
+    dispatch(testProcessFromJson(processId, JSON.stringify({"sourceId":selectedSource, "record": recordJson}), processToDisplay))
```

Comment:
Maybe it should be better to create test data on backend? :thinking: Wasn't sure

Reply:
Not maybe - for sure! :)